<launch>
  <arg name="car" default="true"/>
  <arg name="pedestrian" default="false"/>
  <arg name="sync" default="false" />

  <!-- obj_fusion for car -->
  <group if="$(arg car)">
    <group ns="obj_car">

      <node pkg="lidar_tracker" name="fusion" type="obj_fusion">
        <remap from="/obj_car/obj_label" to="/sync_obj_fusion/obj_car/obj_label" if="$(arg sync)" />
        <remap from="/cluster_centroids" to="/sync_obj_fusion/cluster_centroids" if="$(arg sync)" />
      </node>

    </group>
  </group>

  <!-- obj_fusion for pedestrian -->
  <group if="$(arg pedestrian)">
    <group ns="obj_person">

      <node pkg="lidar_tracker" name="fusion" type="obj_fusion">
        <remap from="/obj_person/obj_label" to="/sync_obj_fusion/obj_person/obj_label" if="$(arg sync)" />
        <remap from="/cluster_centroids" to="/sync_obj_fusion/cluster_centroids" if="$(arg sync)" />
      </node>

    </group>
  </group>

</launch>
